addons/calendar/views/calendar_views.xml

```diff
@@ -375,6 +376,7 @@
                 <field name="display_description" invisible="1"/>
                 <field name="description" invisible="not display_description" options="{'icon': 'fa fa-bars'}"/>
                 <field name="privacy" invisible="1"/>
+                <field name="res_model_name" invisible="not res_model_name" options="{'icon': 'fa fa-link', 'action': 'action_open_calendar_event'}"/>
```

Comment:
Instead of having to set an action, you could have just a boolean like `shouldOpenRecord`, but again just another way of doing things. I'm not particularly against the way you did it